### DISCUSSION TITLE
Mass distance query

### DISCUSSION THREAD

**Discussion by jpmaterial** [shapely/shapely#1261] [Q&A] [answered]
Is there a way to do mass distance calculation other than a for-loop in python? Something like

    p = Polygon(...)
    mp = MultiPoint([(0,0), ..., (100,100)])
    d = p.multi_distance( mp ) # d is a list

or

    p = Polygon(...)
    d = p.multi_distance( np.array( [ (0,0), ..., (100,100) ] ) ) # d is an array

?

**Comment by jorisvandenbossche** [comment, marked as answer]
Starting with Shapely 2.0, this will be possible, as we are adding such "vectorized" functions (moving the functionality of https://github.com/pygeos/pygeos into Shapely).  
Using the current development version (which will become Shapely 2.0), you can do:

```python
>>> import shapely
>>> from shapely.geometry import Polygon

>>> poly = Polygon([(0, 0), (1, 0), (1, 1), (0, 1)])
>>> points = shapely.points(np.array([(0, 0), (1, 1), (100, 100)]))
>>> points
array([<shapely.geometry.Point POINT (0 0)>,
       <shapely.geometry.Point POINT (1 1)>,
       <shapely.geometry.Point POINT (100 100)>], dtype=object)

>>> shapely.distance(poly, points)
array([  0.        ,   0.        , 140.00714267])
```
